blueflood-udp/src/main/java/com/rackspacecloud/blueflood/service/udp/UDPMetricSerialization.java

```diff
@@ -105,26 +107,44 @@ private static int computeBinarySize(Collection<Metric> metrics) {
         
         return size;
     }
+
     private static void write(Metric metric, CodedOutputStream out) throws IOException {
-        
+
         out.writeStringNoTag(metric.getLocator().getTenantId());
         out.writeStringNoTag(metric.getLocator().getMetricName());
-        out.writeStringNoTag(metric.getDataType().toString());
+        DataType dataType = getDataType(metric);
+
+        out.writeStringNoTag(dataType.toString());
         out.writeRawVarint64(metric.getCollectionTime());
         out.writeRawVarint32(metric.getTtlInSeconds());
         out.writeStringNoTag(metric.getUnit());
         if (metric.getDataType().equals(DataType.STRING))
             out.writeStringNoTag((String) metric.getMetricValue());
-        else if (metric.getDataType().equals(DataType.INT))
+        else if (dataType.equals(DataType.INT))
             out.writeRawVarint32((Integer)metric.getMetricValue());
-        else if (metric.getDataType().equals(DataType.LONG))
-            out.writeRawVarint64((Long)metric.getMetricValue());
-        else if (metric.getDataType().equals(DataType.DOUBLE))
+        else if (dataType.equals(DataType.LONG))
+            out.writeRawVarint64((Long) metric.getMetricValue());
+        else if (dataType.equals(DataType.DOUBLE))
             out.writeDoubleNoTag((Double) metric.getMetricValue());
         else if (metric.getDataType().equals(DataType.BOOLEAN))
             out.writeBoolNoTag((Boolean) metric.getMetricValue());
     }
-    
+
+    private static DataType getDataType(Metric metric) {
```

Comment:
We can remove this block of code as well, I think. and then use `instanceOf` operator to get the type of the object in the `write` block.